rlbot/src/render.rs

```diff
@@ -0,0 +1,208 @@
+/// draw.line_3d(car.pos, car.pos + car.upward() * 120., BLUE);
+/// packet_queue.push(draw.build());
+/// ```
+pub struct Renderer {
+    pub group: RenderGroup,
+}
+
+impl Renderer {
+    /// Create a new Renderer.
+    /// Each render group must have a unique id.
+    /// Re-using an id will result in overwriting (watch out when using hiveminds).
+    pub fn new(group_id: i32) -> Self {
+        Self {
+            group: RenderGroup {
+                render_messages: vec![],
+                id: group_id,
+            },
+        }
+    }
+
+    /// Get the resulting [RenderGroup].
+    pub fn build(self) -> RenderGroup {
+        self.group
+    }
+
+    /// Add a [RenderMessage] to this group.
+    pub fn push(&mut self, message: impl Into<RenderMessage>) {
+        self.group.render_messages.push(message.into());
+    }
+
+    /// Draws a line between two anchors in 3d space.
+    pub fn line_3d(
+        &mut self,
+        start: impl Into<RenderAnchor>,
+        end: impl Into<RenderAnchor>,
+        color: Color,
+    ) {
+        self.group.render_messages.push(
+            Line3D {
+                start: Box::new(start.into()),
+                end: Box::new(end.into()),
+                color,
+            }
+            .into(),
+        );
+    }
+
+    /// Draws a line going through each of the provided points.
+    pub fn polyline_3d(
+        &mut self,
+        points: impl IntoIterator<Item = impl Into<Vector3>>,
+        color: Color,
+    ) {
+        self.group.render_messages.push(
+            PolyLine3D {
+                points: points.into_iter().map(|p| p.into()).collect(),
+                color,
+            }
+            .into(),
+        );
+    }
+
+    /// Draws text in 2d space.
+    /// X and y uses screen-space coordinates, i.e. 0.1 is 10% of the screen width/height.
+    /// Use `set_resolution` to change to pixel coordinates.
+    /// Characters of the font are 20 pixels tall and 10 pixels wide when `scale == 1.0`.
+    /// Consider using [push] and `..default()` when using multiple default values.
+    pub fn string_2d(
+        &mut self,
+        text: impl Into<String>,
+        x: f32,
+        y: f32,
+        scale: f32,
+        foreground: Color,
+        background: Color,
+        h_align: TextHAlign,
+        v_align: TextVAlign,
+    ) {
+        self.group.render_messages.push(
+            String2D {
+                text: text.into(),
+                x,
+                y,
+                scale,
+                foreground,
+                background,
+                h_align,
+                v_align,
+            }
+            .into(),
+        );
+    }
+
+    /// Draws text anchored in 3d space.
+    /// Characters of the font are 20 pixels tall and 10 pixels wide when `scale == 1.0`.
+    /// Consider using [push] and `..default()` when using multiple default values.
+    pub fn string_3d(
+        &mut self,
+        text: impl Into<String>,
```

Comment:
I think we should just do this here and above in `string_2d`
```suggestion
        text: String,
```